luci_grpc_interface/interface/src/interface.cpp

```diff
@@ -376,30 +378,47 @@ void Interface::processRearIrData()
 void Interface::sendJsCallback(const luci_messages::msg::LuciJoystick::SharedPtr msg)
 {
     // Send the remote JS values over gRPC
-    spdlog::debug("Received js val: {} {}", msg->forward_back, msg->left_right);
+    auto inputSource = static_cast<InputSource>(msg->input_source);
+    spdlog::debug("Received js val: {} {} {}", msg->forward_back, msg->left_right,
+                  msg->input_source);
 
-    this->luciInterface->sendJS(msg->forward_back + 100, msg->left_right + 100);
+    this->luciInterface->sendJS(msg->forward_back + 100, msg->left_right + 100, inputSource);
 }
 
-void Interface::switchLuciModeCallback(const luci_messages::msg::LuciDriveMode::SharedPtr msg)
+void Interface::setRemoteInputSource()
 {
-    // Send a mode switch over gRPC
-    spdlog::info("Received mode switch request");
-    switch (msg->mode)
+    int checker = this->luciInterface->setInputSource(InputSource::Remote);
+    std::chrono::milliseconds waitTime(100);
+    if (checker == 0)
     {
-    case luci_messages::msg::LuciDriveMode::USER:
-        this->luciInterface->activateUserMode();
-        break;
-    case luci_messages::msg::LuciDriveMode::ENGAGED:
-        this->luciInterface->activateEngagedMode();
-        break;
-    case luci_messages::msg::LuciDriveMode::AUTO:
-        this->luciInterface->activateAutoMode();
-        break;
-    default:
-        spdlog::error("NOT A VALID MODE SELECTION");
-        break;
+        this->luciInterface->sendJS(100, 100, InputSource::Remote);
+        while (true)
+        {
+            // Make sure that the joysick value is 0,0 before sending other values
```

Comment:
This comment doesn't make sense to me. I'd suggest rephrasing to something like "Send origin joystick values until the source switches to remote. This is because LUCI requires the joystick to be at the origin before actually switching input source."